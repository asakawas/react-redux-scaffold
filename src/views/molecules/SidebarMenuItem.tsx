import React, { ReactElement, FunctionComponent } from 'react';
import ListItem, { ListItemProps } from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText, { ListItemTextProps } from '@material-ui/core/ListItemText';

/**
 * The type of props of SidebarMenuItem.
 */
export type SidebarMenuItemProps = Readonly<{
  /** The text on the menu item. */
  text: ListItemTextProps['primary'];

  /** The menu item icon. */
  icon: ReactElement;

  /** The event handler called when the list item is clicked. */
  onClick: ListItemProps<'div'>['onClick'];
}>;

const SidebarMenuItem: FunctionComponent<SidebarMenuItemProps> = ({ text, icon, onClick }) => (
  <ListItem button dense onClick={onClick}>
    <ListItemIcon>{icon}</ListItemIcon>
    <ListItemText primary={text} primaryTypographyProps={{ variant: 'h6' }} />
  </ListItem>
);

export default SidebarMenuItem;
