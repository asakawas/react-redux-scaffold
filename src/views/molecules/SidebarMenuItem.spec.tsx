import React from 'react';
import { render } from '@testing-library/react';
import SidebarMenuItem from './SidebarMenuItem';

describe('SidebarMenuItem', () => {
  test('is rendered as expected', () => {
    const { asFragment } = render(<SidebarMenuItem text="test" icon={<></>} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("contains a node set to 'icon'", () => {
    const renderResult = render(<SidebarMenuItem text="test" icon={<span>HOGE</span>} />);

    const iconNode = renderResult.getByText('HOGE');
    expect(iconNode).not.toBeNull();
    expect(renderResult.container).toContainElement(iconNode);
  });
});
