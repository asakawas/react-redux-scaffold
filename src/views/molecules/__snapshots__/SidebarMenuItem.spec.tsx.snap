// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarMenuItem is rendered as expected 1`] = `
<DocumentFragment>
  <div
    aria-disabled="false"
    class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-button"
    role="button"
    tabindex="0"
  >
    <div
      class="MuiListItemIcon-root"
    />
    <div
      class="MuiListItemText-root MuiListItemText-dense"
    >
      <span
        class="MuiTypography-root MuiListItemText-primary MuiTypography-h6"
      >
        test
      </span>
    </div>
    <span
      class="MuiTouchRipple-root"
    />
  </div>
</DocumentFragment>
`;
